UM-USElab/gradecraft-development
# Ahoy!

### Status
**READY**

### Description
This branch adds a new utility to track events within the application. Ahoy is highly flexible, while also being quite simple in its approach. To inspect events in the console, use either `Ahoy::Event` or `Ahoy::Visit` as the starting point.

### Todos
- [x] Test tracking in staging
- [x] re-add mailer link tests
- [x] track user_id on visits
- [x] Update/Add Documentation https://github.com/UM-USElab/Developer-Documentation/wiki/Ahoy-Tracking 


### Gem dependencies
- ADDED ahoy_matey
- ADDED ahoy_email

### Migrations
YES

### Steps to Test or Reproduce
Log into the application, confirm that there's a new Ahoy::Visit record. Same for submitting an assignment, getting a grade, logging out, predicting things. 

### Impacted Areas in Application
List general components of the application that this PR will affect:

* All pages. Analytics, phew!

======================
